manager/integration/tests/test_settings.py

```diff
@@ -228,4 +231,98 @@ def test_setting_guaranteed_engine_cpu():
+    setting = client.by_id_setting(SETTING_GUARANTEED_ENGINE_CPU)
+    org_setting = setting
+
+    # Get current running instance managers
+    instance_managers = client.list_instance_manager()
+
+    # Set an invalid value, and it should return error
+    try:
+        client.update(setting, value="xxx")
+    except Exception as e:
+        print("Exception when update "
+              "Guaranteed Engine CPU to xxx", e)
+        pass
+
+    # Update guaranteed engine cpu setting to 0.1
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0.1",
+                                        "100m", instance_managers,
+                                        "Running", True)
+
+    # Update guaranteed engine cpu setting to 0
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0",
+                                        None, instance_managers,
+                                        "Running", True)
+
+    # Update guaranteed engine cpu setting to 0.2
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0.2",
+                                        "200m", instance_managers,
+                                        "Running", True)
+
+    # Update guaranteed engine cpu setting to 10
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "10",
+                                        None, instance_managers,
+                                        "Running", False)
+
+    # Update guaranteed engine cpu setting to 0.25
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0.25",
+                                        "250m", instance_managers,
+                                        "Running", True)
+
+    # Create a volume to test
+    vol_name = generate_volume_name()
+    volume = create_and_check_volume(client, vol_name)
+    volume.attach(hostId=get_self_host_id())
+    volume = wait_for_volume_healthy(client, vol_name)
+    assert len(volume.replicas) == 3
+
+    data = write_volume_random_data(volume)
+    check_volume_data(volume, data)
+
+    # Cleanup volume
+    cleanup_volume(client, volume)
+
+    # restore original setting
+    try:
+        client.update(setting, value=org_setting.value)
+    except Exception as e:
+        print("Exception when update "
+              "Guaranteed Engine CPU to", org_setting.value, e)
+        assert(not e)
+
+
+def guaranteed_engine_cpu_setting_check(client, core_api, setting, val,
+                                        cpu_val, instance_managers,
+                                        state, desire):
+    """
+    We chech if intance managers are in the desired state with
+    correct setting
+    """
+    # Update guaranteed engine cpu setting
+    try:
+        client.update(setting, value=val)
+    except Exception as e:
+        print("Exception when update "
+              "Guaranteed Engine CPU to ", val, e)
+        assert(not e)
+
+    # Wait for the current instance manager running
```

Comment:
Remove this comment. It's obvious from the function name. Also, it's not waiting for it to be `running`.